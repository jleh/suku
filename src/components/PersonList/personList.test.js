import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import { PersonListWithoutContext } from '.';

const renderComponent = (persons) =>
  renderer
    .create(
      <MemoryRouter>
        <PersonListWithoutContext personList={persons} />
      </MemoryRouter>
    )
    .toJSON();

test('Renders nothing if no events', () => {
  const persons = [
    {
      id: 'H1',
      name: 'Test B',
    },
    {
      id: 'H2',
      name: 'Test C',
    },
    {
      id: 'H3',
      name: 'Test A',
    },
    {
      id: 'H4',
      name: 'Test af D',
    },
    {
      id: 'H5',
      name: ' Private',
    },
  ];

  expect(renderComponent(persons)).toMatchSnapshot();
});
