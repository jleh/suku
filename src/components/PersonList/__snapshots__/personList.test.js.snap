// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders nothing if no events 1`] = `
<div>
  <h2>
    Missing translationId: menu.personIndex for language: \${ languageCode }
  </h2>
  <div>
    <a
      href="/person/H3"
      onClick={[Function]}
    >
      Test A
    </a>
  </div>
  <div>
    <a
      href="/person/H1"
      onClick={[Function]}
    >
      Test B
    </a>
  </div>
  <div>
    <a
      href="/person/H2"
      onClick={[Function]}
    >
      Test C
    </a>
  </div>
  <div>
    <a
      href="/person/H4"
      onClick={[Function]}
    >
      Test af D
    </a>
  </div>
</div>
`;
